### PR TITLE
[Mail]: Add a new event to modify data before persisting

### PR DESCRIPTION
**Pull request by MarkTro**
For privacy policy it is useful to have a possibility to modify the mail data before it will persist in the database.

### CODE REVIEW

File: lib/Mail.php [hunk 1]
```diff
@@ -553,6 +553,8 @@ public function sendWithoutRendering(MailerInterface $mailer = null): static
                 $recipients = $this->getDebugMailRecipients($recipients);
             }
 
+            \Pimcore::getEventDispatcher()->dispatch($event, MailEvents::PRE_LOG);
```
Comment (jdreesen) [inline lib/Mail.php:556]: I'm not sure if we really should re-use the same object here :thinking:

Comment (MarkTro) [reply lib/Mail.php:556]: It's that same pattern as always within event dispatcher. Why would you do it different in this case?

Comment (robertSt7) [reply lib/Mail.php:556]: @jdreesen I agree with @MarkTro

**Comment by robertSt7** [comment]
@MarkTro Hi the PR looks good, but I am not sure how we can now modify the data before we persist it. Could you please provide an code example? Thanks

**Comment by MarkTro** [comment]
Hi @robertSt7 ,
no problem:
```
class ModifyMailSubscriber implements EventSubscriberInterface
{
   public static function getSubscribedEvents(): array
    {
        return [
            MailEvents::PRE_LOG => 'process',
        ];
    }

    public function process(MailEvent $event): void
    {
        $mail = $event->getMail();

        //Modify html body
        $html = $mail->getHtmlBody();
        $newHtml = preg_replace('/patternOfContentToRelace/', 'newContent', $html);
        $mail->setHtmlBody($newHtml);

        //Modify text body
        $text = $mail->getTextBody();
        $newText = preg_replace('/patternOfContentToRelace/', 'newContent', $text);
        $mail->setTextBody($newText);
    }
}
```

**Comment by robertSt7** [comment]
@MarkTro Thanks for the improvement

**Pull request merged**